modules/examples/sources/msg-generator/msg-generator-source.c

```diff
@@ -93,25 +94,39 @@ _free(LogPipe *s)
   log_source_free(s);
 }
 
+static void
+_add_name_value(gpointer key, gpointer value, gpointer data)
+{
+  gchar *name = (gchar *) key;
+  LogTemplate *val = (LogTemplate *) value;
+  MsgGeneratorSource *self = (MsgGeneratorSource *) data;
+  GString *msg_body = g_string_sized_new(128);
+  log_template_format(val, self->msg, NULL, LTZ_LOCAL, 0, NULL, msg_body);
+  log_msg_set_value_by_name(self->msg, name, msg_body->str, msg_body->len);
+  g_string_free(msg_body, TRUE);
+
+}
+
 static void
 _send_generated_message(MsgGeneratorSource *self)
 {
   if (!log_source_free_to_send(&self->super))
     return;
 
-  LogMessage *msg = log_msg_new_empty();
-  log_msg_set_value(msg, LM_V_MESSAGE, "-- Generated message. --", -1);
+  self->msg = log_msg_new_empty();
```

Comment:
I used it as self because I am using it in another function when I'm assigning the name value pairs(Basically for g_hash_table_foreach). How else can I pass the value?

Reply:
Instead of passing `self` to foreach, you can pass `msg` directly as `user_data`. There is no need to pass the entire source.

Reply:
I had passed message earlier as user_data, But then there was memory leak when it came to hashmap value. I found this as a fix to that and it works fine

Reply:
Alright